xain_fl/coordinator/metrics_store.py

```diff
@@ -0,0 +1,127 @@
+"""XAIN FL Metric Store"""
+
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta
+from typing import Dict, List
+
+from influxdb import InfluxDBClient
+from numpy import array2string, ndarray
+
+from xain_fl.config import MetricsConfig
+from xain_fl.logger import StructLogger, get_logger
+
+logger: StructLogger = get_logger(__name__)
+
+
+class AbstractMetricsStore(ABC):  # pylint: disable=too-few-public-methods
+    """An abstract metric store."""
+
+    @abstractmethod
+    def write_metrics(self, participant_id: str, metrics: Dict[str, ndarray]) -> bool:
+        """Write the participant metrics on behalf of the participant with the given participant_id
+        into a metric store.
+
+        Args:
+
+            participant_id: The ID of the participant.
+            metrics: The metrics of the participant with the given participant_id.
+
+        Returns:
+
+            True, on success, otherwise False.
```

Comment:
I feel like it's a bit weird in python to use a boolean for reporting a result. Do you think we could use a custom `MetricsStoreError` exception instead?

Reply:
Good point, I will adjust it tomorrow.

Reply:
I have adjusted the code